include/qml_ros2_plugin/publisher.hpp

```diff
@@ -51,6 +58,8 @@ class Publisher : public QObjectRos2
 
   quint32 queueSize() const;
 
+  void setQueueSize( quint32 value );
```

Comment:
I've kept the name `QueueSize` here even though I believe it should be `HistoryDepth`?

Reply:
> Yes, I've kept the name `queueSize` for backwards compatibility. Not sure if it makes sense to have a cut there and use the new terminology.
> It's not like these libraries are out for that long so making such a change is still possible.

so change to `HistoryDepth`, or keep `QueueSize`?

> However, the publisher was not really intended to be modified after creation.
> The only reason `Subscription` supports modification is because it can be instantiated as a QML item.
> The publisher receives all required information in the constructor to ensure that it will be in a consistent state from the start.
> Not sure what the use case would be to change such properties dynamically.

things will only get less complex if it can't be changed, so I'd be in favour of removing the setter(s).

Reply:
> so change to HistoryDepth, or keep QueueSize?

or remove it completely and add properties to `QoS` to get these values.
However, QoS is currently a `Q_GADGET` which means it does not support signals so no changed notify signals.
I don't think they are necessary for the `QoS` settings, though, and don't justify dealing with the complications that would arise when changing `QoS` to `Q_OBJECT`.

Reply:
As I'm not a Qt aficionado I'm going to need some guidance in this case.

If you'd rather have properties in `QoS`, that would be fine by me.

Reply:
Currently, `QoS` is a `Q_GADGET` class. The difference to a standard `QObject` based class is that we don't have all that `QObject` overhead at the cost of not being connected to the signal mechanism.
Hence, we can have properties (I think) but they can't notify of changes since the `Q_GADGET` can't have signals.
I don't think we need them though and switching to `QObject` would make building the `QoS` more complicated since the gadget is passed by value whereas `QObject` is always passed by pointer and I'm currently using the value mechanism to enable that you can start configuring QoS use them for one publisher and add some more settings for the second one without these changes being reflected on the first publisher. I think that behavior is closer to the cpp version.

Reply:
So, conclusion: keep `QoS` a `Q_GADGET` class, but add (more) properties to it?

Reply:
Yes, I think adding properties for the available options would make most sense.
Since humble there are getters for properties such as the `history` but they have the same name as the setter.
I don't think that will work with QML. Will have to get around that.